### PR TITLE
Feature/staking issue

### PR DESCRIPTION
**Pull request by shortthefomo**
lathans attempt to fix some of the stacking issues.
patching

-  missing awaits
- _fetching lock
- convert setInterval to setTImeout

### CODE REVIEW

File: src/processors/peersync/index.ts [hunk 1]
```diff
@@ -21,6 +21,9 @@ const pollAllPeers = async (peerManager: PeerManager): Promise<void> => {
         await peerManager.poll(peer)
       }
     }
+
+    // induce a loop to its self after timeout has completed.
+    await setTimeout(pollAllPeers, ENV.PEERSYNC_POLL_INTERVAL_MS, peerManager)
```
Comment (pkcs8) [inline src/processors/peersync/index.ts:26]: `setTimeout` does not return Promise. Therefore, this await may not be necessary.

File: src/processors/peersync/PollService.ts [hunk 2]
```diff
@@ -17,12 +17,14 @@ class PollService {
   private readonly _manager: PeerManager
   private _startTime: number
   private _totalDocs: number
+  private _fetching: boolean
```
Comment (pkcs8) [inline src/processors/peersync/PollService.ts:20]: I may be wrong in understanding `_fetching`'s behavior, but this lock will help only when `pollService` is instantiated and `await pollService.fetch()` is called twice.

File: src/processors/peersync/index.ts (hunk 1 above)
Comment (shortthefomo) [reply src/processors/peersync/index.ts:26]: yup agreed

**Comment by shortthefomo** [comment]
closing this PR out in favour of breaking it up into 3 commits.

**Pull request closed without merging**